### DISCUSSION TITLE
Background task?

### DISCUSSION THREAD

**Discussion by magnusjjj** [ClearFoundry/ClearScript#280] [Q&A]
I have a script, where my function, 'async Initialize()' opens a websocket in v8, using System.Net.WebSockets.ClientWebSocket
ClientWebSocket is async, so in the script code, I need to await this.ws.ReceiveAsync.
This all works, and happily outputs all the messages.
But it blocks when I run it.

I put all these messages I get from the websocket into a list, and I am trying to read them from another script function, UpdateAfter().

But, because the first call to Initialize() blocks, UpdateAfter() never runs, and in fact if I call it from C# the whole program stops, presumably because the V8 engine is locked.

Is there a way to do a producer / consumer setup like that, at all? :)

**Comment by ClearScriptLib** [comment]
Hi @magnusjjj,

What you're describing should be possible, and we test many such scenarios.

>But it blocks when I run it.

What exactly is blocking? Can you provide more information, or a minimal sample that reproduces the issue?

Thanks!
